packages/keys/src/key-pair.ts

```diff
@@ -21,5 +24,26 @@ export async function createKeyPairFromBytes(bytes: Uint8Array, extractable?: bo
         crypto.subtle.importKey('raw', bytes.slice(32), 'Ed25519', /* extractable */ true, ['verify']),
         createPrivateKeyFromBytes(bytes.slice(0, 32), extractable),
     ]);
+
+    // Verify the key pair
+    const signedData = await signBytes(privateKey, TEST_DATA);
+    const isValid = await verifySignature(publicKey, signedData, TEST_DATA);
+    if (!isValid) {
+        throw new SolanaError(SOLANA_ERROR__KEYS__VERIFY_SIGNATURE_FAILURE);
+    }
+
+    return { privateKey, publicKey } as CryptoKeyPair;
+}
+
+// Similar to createKeyPairFromBytes, but does not verify the key pair
+export async function createKeyPairFromBytes_DANGEROUSLY_SKIP_VALIDATION(bytes: Uint8Array, extractable?: boolean): Promise<CryptoKeyPair> {
+    if (bytes.byteLength !== 64) {
+        throw new SolanaError(SOLANA_ERROR__KEYS__INVALID_KEY_PAIR_BYTE_LENGTH, { byteLength: bytes.byteLength });
+    }
+    const [publicKey, privateKey] = await Promise.all([
+        crypto.subtle.importKey('raw', bytes.slice(32), 'Ed25519', /* extractable */ true, ['verify']),
+        createPrivateKeyFromBytes(bytes.slice(0, 32), extractable),
+    ]);
+
     return { privateKey, publicKey } as CryptoKeyPair;
 }
```

Comment:
I know the original issue suggested we expose such function but IMO we shouldn't.

If the user is looking to tradeoff that layer of security for performance, it's a very small function to copy/paste.